hls4ml/backends/vivado/vivado_backend.py

```diff
@@ -375,10 +375,38 @@ def init_depconv2d(self, layer):
 
     def _set_pooling_accum_t(self, layer, pool_size):
         extra_bits = ceil_log2(pool_size)
+        if 'inputs' in layer.attributes:
+            # Functional
+            input_layer_name = layer.attributes['inputs'][0]
+        else:
+            # Sequential
```

Comment:
I don't fully understand why the if-else is requried. Can't you always query for the previous node?

Reply:
If we have a functional model that has A->B, A->C, with C being the pooling layer. As the order of B and C is arbitrary, if we query the precious node, we can get B, the sibling of the current node, instead of A.

Reply:
I don't quite follow. Are the node inputs not correctly filled in some cases?

Reply:
Not sure for now, but they were not as of Nov 2023. Only for the functional model converted graph, the parent names are filed; for sequential models, the fields were missing.
Could check this again quickly.

Reply:
![1](https://github.com/fastmachinelearning/hls4ml/assets/29270722/c658c617-92fd-48fe-a23a-6243aeb60f92)
![2](https://github.com/fastmachinelearning/hls4ml/assets/29270722/9c1133d7-4e94-4361-8e92-03403e81c9ff)
Confirmed.

Reply:
But look at the jettagging from the hls4ml-tutorial:
```python
model = Sequential()
model.add(Dense(64, input_shape=(16,), name='fc1', kernel_initializer='lecun_uniform', kernel_regularizer=l1(0.0001)))
model.add(Activation(activation='relu', name='relu1'))
model.add(Dense(32, name='fc2', kernel_initializer='lecun_uniform', kernel_regularizer=l1(0.0001)))
model.add(Activation(activation='relu', name='relu2'))
model.add(Dense(32, name='fc3', kernel_initializer='lecun_uniform', kernel_regularizer=l1(0.0001)))
model.add(Activation(activation='relu', name='relu3'))
model.add(Dense(5, name='output', kernel_initializer='lecun_uniform', kernel_regularizer=l1(0.0001)))
model.add(Activation(activation='softmax', name='softmax'))
```
and looking at the node object for `fc2` you can see that the inputs are `['relu1]`:

Reply:
<img width="526" alt="image" src="https://github.com/fastmachinelearning/hls4ml/assets/6638210/5526b74a-fa1b-478b-bec3-1b4c5101c442">

Reply:
So you can use `get_input_node` or something similar to get the node. https://github.com/fastmachinelearning/hls4ml/blob/main/hls4ml/model/layers.py#L182-L192

Reply:
Just checked and cannot get the inputs with current hls4ml master.... weird.

Reply:
Note that it's `self.inputs`, not `self.attributes['inputs']`.

Reply:
(on the Layer object)

Reply:
Confirmed, modified PR.